android/src/main/java/com/pspdfkit/flutter/pspdfkit/ConfigurationAdapter.java

```diff
@@ -0,0 +1,370 @@
+    private static final String FIT_PAGE_TO_WIDTH = "fitPageToWidth";
+    private static final String IMMERSIVE_MODE = "immersiveMode";
+    private static final String USER_INTERFACE_VIEW_MODE = "userInterfaceViewMode";
+    private static final String USER_INTERFACE_VIEW_MODE_AUTOMATIC = "automatic";
+    private static final String USER_INTERFACE_VIEW_MODE_AUTOMATIC_BORDER_PAGES = "automaticBorderPages";
+    private static final String USER_INTERFACE_VIEW_MODE_ALWAYS_VISIBLE = "alwaysVisible";
+    private static final String USER_INTERFACE_VIEW_MODE_ALWAYS_HIDDEN = "alwaysHidden";
+    private static final String ANDROID_SHOW_SEARCH_ACTION = "showSearchAction";
+    private static final String INLINE_SEARCH = "inlineSearch";
+    private static final String SHOW_THUMBNAIL_BAR = "showThumbnailBar";
+    private static final String SHOW_THUMBNAIL_BAR_DEFAULT = "default";
+    private static final String SHOW_THUMBNAIL_BAR_SCROLLABLE = "scrollable";
+    private static final String SHOW_THUMBNAIL_BAR_NONE = "none";
+    private static final String ANDROID_SHOW_THUMBNAIL_GRID_ACTION = "showThumbnailGridAction";
+    private static final String ANDROID_SHOW_OUTLINE_ACTION = "showOutlineAction";
+    private static final String ANDROID_SHOW_ANNOTATION_LIST_ACTION = "showAnnotationListAction";
+    private static final String SHOW_PAGE_NUMBER_OVERLAY = "showPageNumberOverlay";
+    private static final String SHOW_PAGE_LABELS = "showPageLabels";
+    private static final String INVERT_COLORS = "invertColors";
+    private static final String GRAY_SCALE = "grayScale";
+    private static final String START_PAGE = "startPage";
+    private static final String ENABLE_ANNOTATION_EDITING = "enableAnnotationEditing";
+    private static final String ENABLE_TEXT_SELECTION = "enableTextSelection";
+    private static final String ANDROID_SHOW_SHARE_ACTION = "showShareAction";
+    private static final String ANDROID_SHOW_PRINT_ACTION = "showPrintAction";
+    private static final String SHOW_DOCUMENT_INFO_VIEW = "showDocumentInfoView";
+    private static final String APPEARANCE_MODE = "appearanceMode";
+    private static final String APPEARANCE_MODE_DEFAULT = "default";
+    private static final String APPEARANCE_MODE_NIGHT = "night";
+    private static final String ANDROID_DARK_THEME_RESOURCE = "darkThemeResource";
+    private static final String ANDROID_DEFAULT_THEME_RESOURCE = "defaultThemeResource";
+
+    private final PdfActivityConfiguration.Builder configuration;
+
+    @SuppressWarnings("ConstantConditions")
+    ConfigurationAdapter(@NonNull Context context,
+                         @Nullable HashMap<String, Object> configurationMap) {
+        if (configurationMap == null || configurationMap.isEmpty()) {
+            this.configuration = getDefaultConfiguration(context);
+        } else {
+            this.configuration = new PdfActivityConfiguration.Builder(context);
```

Comment:
I think we should use `getDefaultConfiguration(context)` for creating the configuration of this branch too. Right now, the default options differ based on whether no configuration was passed, or a partial configuration was provided. Instead, the default configuration should be the same for both cases, and passed in options should only override the defaults. It would probably make sense to always use `getDefaultConfiguration(context)` and then use this builder for both branches.

Reply:
The reason behind this was to keep it similar to the react-native wrapper we currently have implemented https://github.com/PSPDFKit/react-native/blob/master/android/src/main/java/com/pspdfkit/react/ConfigurationAdapter.java#L69-L72
I would change it in both or none.

Reply:
I would argue that the current behavior does not make sense (neither here, nor in our RN wrapper).

If you consider these two configurations, what would you expect the difference to be?

1) 
```dart
Pspdfkit.present(..., {
        pageScrollDirection: HORIZONTAL
});
```

2)
```dart
Pspdfkit.present(..., {});
```

Considering that horizontal scroll direction is the default, I would assume that these two configurations are identical. Still, they show a handful of differences (1) uses always inline search, whereas 2) selects depending on tablet/phone, 1) uses fit to width vs. 2) fit to screen, etc.). More importantly, these differences will even be different per platform (since they come from the platforms). We should make sure our wrappers use the same defaults across platforms too, where possible.